### ISSUE TITLE
False positive no-member

### ISSUE THREAD

**Issue by fluffy-critter** [PyCQA/pylint#4693]
### Steps to reproduce

In https://github.com/PlaidWeb/Authl/blob/dc53e065e7cfe5e21bcfb169aea4588293531da8/tests/handlers/test_fediverse.py#L89:

```
    result = handler.check_callback(result.url, parse_args(result.url), {})
    assert isinstance(result, disposition.Verified)
    assert result.identity == 'https://mastodon.example/@moo'
    assert result.redir == 'qwerpoiu'
    assert result.profile == { 
        'name': 'moo friend',
        'bio': 'a cow',
        'avatar': 'https://placekitten.com/1280/1024',
        'homepage': 'https://moo.example',
        'pronouns': 'moo/moo'
    }

```

### Current behavior

```
tests/handlers/test_fediverse.py:93:11: E1101: Instance of 'Error' has no 'profile' member (no-member)
```

### Expected behavior

In this particular test, `result` can never be `disposition.Error` and there's even a previous assertion to ensure that it's [`disposition.Verified`](https://github.com/PlaidWeb/Authl/blob/dc53e065e7cfe5e21bcfb169aea4588293531da8/authl/disposition.py#L35) which does have the `'profile'` member. I assume that there's some erroneous caching of `result`'s type from another test.

### pylint --version output

Result of `pylint --version` output:

```
pylint 3.0.0-a4
astroid 2.6.2
Python 3.8.6 (default, Jan  6 2021, 12:14:27) 
[Clang 12.0.0 (clang-1200.0.32.28)]
```

**Comment by jolaf** [comment]
I've encountered this error too – twice already.
